lib/actions/rootActions.js

```diff
@@ -274,10 +292,24 @@ export const stopMoveChart = () => ({
   type: actions.STOP_MOVE_CHART
 });
 
-export const deleteChart = index => ({
-  type: actions.DELETE_CHART,
-  index
-});
+export const deleteChart = index => (dispatch, getState) => {
+  const approvalDelChart = confirm('Do You want to delete this chart?');
+  if (approvalDelChart) {
+    const store = getState();
+    const dashboardInfo = getDashboardInfo(store);
+    const delValue = dashboardInfo.data.items[index].savedQuery;
+    dispatch(changeSavedQueryList(delValue, []));
+    const newStore = getState();
+    const savedQueriesList = getSavedQueriesList(newStore);
+    delValue.some(sq => !savedQueriesList.includes(sq)) &&
```

Comment:
The checking condition could start from boolean value `dashboardInfo.is_public` to avoid unnecessary array looping.